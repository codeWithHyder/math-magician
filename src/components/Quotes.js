import React, { useState, useEffect } from 'react';
import apiKey from './APIkey';

const Quotes = () => {
  const [data, setData] = useState(null);
  const [error, setError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=education', {
          headers: { 'X-Api-Key': apiKey },
        });
        const res = await response.json();
        setData(res[0]);
      } catch (error) {
        setError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, []);

  if (isLoading) return <div>Loading...</div>;
  if (error) return <div>Something went wrong</div>;
  return (
    <>
      <p id="par">hello</p>
      <ul className="quotes" data-testid="quotes">
        {data && (
        <p>
          {data.quote}
          {' '}
          <span>
            ~
            {data.author}
          </span>
        </p>
        )}
      </ul>
    </>
  );
};

export default Quotes;
