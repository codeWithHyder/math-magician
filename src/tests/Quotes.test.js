import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

test('Quotes rendering', () => {
  render(<Quotes />);

  const title = screen.getByTestId('quotes');

  expect(title).toBeInTheDocument();
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
