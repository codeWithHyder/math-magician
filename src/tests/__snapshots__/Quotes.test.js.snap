// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes rendering 1`] = `
<ul
  className="quotes"
/>
`;
